### PR TITLE
Add prow linter to master

### PR DESCRIPTION
**Pull request by howardjohn**

### CODE REVIEW

**Comment by howardjohn** [comment]
/hold
until https://github.com/istio/istio/pull/14934 merges

**Comment by howardjohn** [comment]
/hold-cancel

@icygalz this is ready now that the test was checked in. Thanks!

**Comment by icygalz** [comment]
Please make circleci:lint to be not required. If not, the failure will
block everybody from merging...

File: prow/cluster/jobs/istio/istio/istio.istio.master.yaml [hunk 1]
```diff
@@ -47,6 +47,21 @@ presubmits:
       nodeSelector:
         testing: test-pool
 
+  - name: istio-lint-master
```
Comment (rshriram) [inline prow/cluster/jobs/istio/istio/istio.istio.master.yaml:50]: make circleci lint as non required

**Comment by howardjohn** [comment]
I would prefer to see a few on prow pass then make prow required and remove circle. Otherwise we will block everyone if prow doesn't work and everyone will have to rebase their PRs to pick up the new test which is a pain

**Comment by icygalz** [comment]
The problem is that circleci:lint test after your istio/istio merge will
always be failing, right?

That will affect other people merging because circleci:lint is a required
test. We need to make circleci:lint not required..

**Comment by howardjohn** [comment]
I don't think my change impacts circle test, it has always been flaky and
sometimes gets OOMkilled.

Unless the tiny junit change uses more memory and pushed it over the edge
consistently now?

I

**Comment by icygalz** [comment]
Oo okay, I thought it would always be failing since admin needs to override
the failure to merge #14934..

/lgtm

**Comment by howardjohn** [comment]
it shouldn't, but if it does we should remove it from required sooner

**Pull request merged**